### DISCUSSION TITLE
Map creation fails - how to proceed?

### DISCUSSION THREAD

**Discussion by marfrh** [mapsforge/mapsforge#1416] [Q&A] [answered]
Hi,

I'm trying to render Italy Centro from Geofabrik [Link](https://download.geofabrik.de/europe/italy/centro-latest.osm.pbf) with latest osmosis 0.48.3 and latest mapwriter. Mapwriter always fails at the same point after 40% of default zoom interval three (see log excerpt attached), and I suspect some large things in osm data to cause the problems (though I already filtered many relation types). 

My question is: How can I find out which geometries mapwriter complains about? `Large geometry XYZ...` --> is `XYZ` an ID and how can I map this ID to osm data/IDs? I could not find any info about this.

Thank you!

[mapwriter_output.txt](https://github.com/mapsforge/mapsforge/files/11902080/mapwriter_output.txt)

**Comment by devemux86** [comment]
Maybe you need to give more memory to the Java process with [-Xmx](https://docs.oracle.com/cd/E13150_01/jrockit_jvm/jrockit/jrdocs/refman/optionX.html) option.

On Mapsforge server we use [mapsforge-creator](https://github.com/mapsforge/mapsforge-creator) with [Osmosis 0.47](https://bretth.dev.openstreetmap.org/osmosis-build/) and Java 11.
The italy/centro.map is built successfully: https://download.mapsforge.org/maps/v5/europe/italy/

**Reply by marfrh** [reply]
Thank you for your reply. 
Regarding Memory: I already set `-Xmx26G` --> could this still be too low for this not too large region?  
As mapwriter does not quit because of OOM, I suspected a different root cause (it quits with a `BufferOverflowException` caused in `writeWayNodes`, so I thought something might be too large).

Regarding my question above: Is there any way to map `XYZ`  from the logs `Large geometry XYZ...` to an osm ID? 

I'll also try to render the unmodified Geofabrik file (without filtering, without land/sea/contourlines, without custom tag mapping) with osmosis 0.47.

**Reply by devemux86** [reply, marked as answer]
I made a local test on a old laptop with the configuration mentioned [above](https://github.com/mapsforge/mapsforge/discussions/1416#discussioncomment-6311973) and the map creation completes successfully.
But I use the original pbf from Geofabrik.

If you make changes to the osm files, you could debug why the map creation fails.
You can edit the map-writer code to add custom logging and produce new jar files.

If you just want to not include some map elements, it is better to use:
- A custom [tag-mapping](https://github.com/mapsforge/mapsforge/blob/master/docs/Getting-Started-Map-Writer.md#defining-a-custom-tag-mapping-via-xml)
- Or [TagTransform](https://wiki.openstreetmap.org/wiki/Osmosis/TagTransform) plugin

**Reply by marfrh** [reply]
Map creation with the unmodified pbf from Geofabrik succeeded also for me (with osmosis 0.47.4 and 0.48.3), so there seems to be a problem in my adjusted osm data.
Thank you for your support!
